data_steward/cdr_cleaner/cleaning_rules/create_expected_ct_list.py

```diff
@@ -0,0 +1,160 @@
+-- may be removed once the program finalizes the inclusion of AI/AN participants --
+, aian_persons as (
+  select distinct person_id
+  from `{{project_id}}.{{dataset_id}}.observation`
+  where observation_source_concept_id IN (1586140) AND value_source_concept_id IN (1586141)
+)
+
+-- get list of all participants who have completed The Basics survey --
+, has_the_basics as (
+     SELECT
+    distinct person_id
+  FROM `{{project_id}}.{{dataset_id}}.concept_ancestor`
+  INNER JOIN `{{project_id}}.{{dataset_id}}.observation` o ON observation_concept_id = descendant_concept_id
+  INNER JOIN `{{project_id}}.{{dataset_id}}.concept` d ON d.concept_id = descendant_concept_id
+  WHERE ancestor_concept_id = 1586134
+
+  UNION DISTINCT
+  SELECT
+   distinct person_id
+  FROM `{{project_id}}.{{dataset_id}}.concept`
+  JOIN `{{project_id}}.{{dataset_id}}.concept_ancestor`
+    ON (concept_id = ancestor_concept_id)
+  JOIN `{{project_id}}.{{dataset_id}}.observation`
+    ON (descendant_concept_id = observation_concept_id)
+  WHERE concept_class_id = 'Module'
+    AND concept_name IN ('The Basics')
+    AND questionnaire_response_id IS NOT NULL
+)
+
+-- get list of all participants who have not completed The Basics survey --
+-- can be used during debugging to determine who doesn't have the basics --
+, missing_the_basics as (
+  SELECT distinct person_id
+  FROM rdr_persons
+  WHERE person_id not in (SELECT distinct person_id FROM has_the_basics)
+)
+ -- get list of all participants that could be dropped due to bad birth date records --
+, bad_birthdate_records as (
+    SELECT person_id 
+    FROM `{{project_id}}.{{dataset_id}}.person` p 
+    WHERE p.year_of_birth < 1800 
+    OR p.year_of_birth > (EXTRACT(YEAR FROM CURRENT_DATE()) - 17)
+)
+
+-- store the research_id of all participants --
+select m.research_id
+from rdr_persons p
+left join `{{project_id}}.{{pipeline_lookup_tables}}.{{mapping_table}}` m
+using(person_id)
+where person_id in (select person_id from has_the_basics)
+AND person_id not in (select person_id from bad_birthdate_records)
+);""")
+
+
+class StoreExpectedCTList(BaseCleaningRule):
+    """
+    Store the expected CT participant list.
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id, namer=None):
+        desc = (
+            f'Creates a sandbox table with expected participant research_ids '
+            f'that will be found in the controlled tier dataset.  May be used '
+            f'to help validate a particular pipeline run.  May be extended to '
+            f'update which service accounts can read the data in the table.')
+
+        super().__init__(
+            issue_numbers=['DC2595'],
+            description=desc,
+            affected_datasets=[],  # has no side effects
+            project_id=project_id,
+            dataset_id=dataset_id,
+            sandbox_dataset_id=sandbox_dataset_id,
+            depends_on=[CleanMappingExtTables],
+            table_namer=namer)
```

Comment:
`WARNING:cdr_cleaner.clean_cdr_engine:StoreExpectedCTList does not accept the ``table_namer`` property yet` is showing in [the CircleCI test warning](https://app.circleci.com/pipelines/github/all-of-us/curation/5876/workflows/2c7dba3f-3612-4a74-b36b-c9fe21ff94d9/jobs/17748/parallel-runs/3?filterBy=ALL&invite=true#step-106-463).

I do not fully understand why, but if we change `table_namer=namer` to `table_namer=table_namer` here and `namer=None` to `table_namer=None` at line 91 will solve that warning message.

Reply:
@hiro-mishima this is happening because of the change on line 220 of data_steward/cdr_cleaner/clean_cdr_engine.py 21 days ago.  The parameter `table_namer` was hard coded and when this rule used `namer` instead of `table_namer` the method signatures did not match, raising the `TypeError` and logging the warning message.  That is why just changing the name fixes the issue.  Does this make sense?